Content.Shared/Xenoarchaeology/Artifact/SharedXenoArtifactSystem.XAT.cs

```diff
@@ -63,6 +64,15 @@ public void TriggerXenoArtifact(Entity<XenoArtifactComponent> ent, Entity<XenoAr
         if (!force && _timing.CurTime < ent.Comp.NextUnlockTime)
             return;
 
+        // Frontier: Disable activations on protected grids
+        var xform = Transform(ent);
+        if (xform.GridUid != null)
+        {
+            if (TryComp<ProtectedGridComponent>(xform.GridUid.Value, out var prot) && prot.PreventArtifactTriggers)
+                return;
+        }
```

Comment:
```suggestion
        if (TryComp(ent, out TransformComponent? xform)
            && TryComp(xform.GridUid, out var prot)
            && prot.PreventArtifactTriggers)
        {
                return;
        }
```
TryComp has nullable versions, they're fine.
Tend to prefer TryComp for TransformComponent vs. Transform(), Transform throws an exception if the entity doesn't have a valid transform (generally true on entities being destroyed?)

This applies to the three uses of Transform you have.  Not a huge deal either way.